httpclient5/src/main/java/org/apache/hc/client5/http/impl/async/AsyncHttpRequestRetryExec.java

```diff
@@ -128,6 +128,9 @@ public void handleInformationResponse(final HttpResponse response) throws HttpEx
             public void completed() {
                 if (state.retrying) {
                     scope.execCount.incrementAndGet();
+                    if (entityProducer != null && entityProducer.isRepeatable()) {
```

Comment:
@JasonMathison Damn. Good catch. However, I propose `AsyncEntityProducer#isRepeatable` check be dropped here. At this point if the entity producer is not null we should always call `AsyncEntityProducer#releaseResources`.

Reply:
PR updated.